daemon/src/projection.rs

```diff
@@ -289,56 +289,54 @@ impl Cfd {
                     network,
                     TxLabel::Lock,
                 );
-                self.details.tx_url_list.push(tx_url);
+                self.details.tx_url_list.insert(tx_url);
                 self.expiry_timestamp = Some(dlc.settlement_event_id.timestamp());
                 self.latest_dlc = Some(dlc);
-
-                (CfdState::PendingOpen, vec![])
+                self.state = CfdState::PendingOpen;
             }
             ContractSetupFailed => {
                 // Don't display profit for failed contracts.
                 self.profit_btc = None;
                 self.profit_percent = None;
-
-                (CfdState::SetupFailed, vec![])
+                self.state = CfdState::SetupFailed;
             }
             OfferRejected => {
                 // Don't display profit for rejected contracts.
                 self.profit_btc = None;
                 self.profit_percent = None;
-
-                (CfdState::Rejected, vec![])
+                self.state = CfdState::Rejected;
             }
             RolloverCompleted { dlc } => {
                 self.expiry_timestamp = Some(dlc.settlement_event_id.timestamp());
                 self.latest_dlc = Some(dlc);
 
-                (CfdState::Open, vec![])
+                self.state = CfdState::Open;
+            }
+            RolloverRejected => {
+                self.state = CfdState::Open;
+            }
+            RolloverFailed => {
+                self.state = CfdState::Open;
             }
-            RolloverRejected => (CfdState::Open, vec![]),
-            RolloverFailed => (CfdState::Open, vec![]),
             CollaborativeSettlementStarted { proposal } => match role {
                 Role::Maker => {
                     self.pending_settlement_proposal_price = Some(proposal.price);
-
-                    (
-                        CfdState::IncomingSettlementProposal,
-                        vec![CfdAction::AcceptSettlement, CfdAction::RejectSettlement],
-                    )
+                    self.state = CfdState::IncomingSettlementProposal;
+                }
+                Role::Taker => {
+                    self.state = CfdState::OutgoingSettlementProposal;
                 }
-                Role::Taker => (CfdState::OutgoingSettlementProposal, vec![]),
             },
             CollaborativeSettlementProposalAccepted => {
                 self.pending_settlement_proposal_price = None;
-
-                (CfdState::IncomingSettlementProposal, vec![])
+                self.state = CfdState::IncomingSettlementProposal;
```

Comment:
I suspect that this is a bug on the maker side -  won't this code will show the buttons to "Accept/Reject" on the maker after accepting a proposal? (we might need to expand the pure function to look at the state + other things, or create more UI-only states....)

Reply:
Uhm true! We don't have an in-between state for that I think!